applications/incompressible_navier_stokes/flow_past_cylinder/include/grid.h

```diff
@@ -1391,18 +1484,38 @@ void
 create_coarse_grid(dealii::Triangulation<dim> &                             triangulation,
                    std::vector<dealii::GridTools::PeriodicFacePair<
                      typename dealii::Triangulation<dim>::cell_iterator>> & periodic_faces,
-                   std::string                                              cylinder_type_string)
+                   std::string                                              cylinder_type_string,
+                   ElementType                                              element_type)
 {
+  if(element_type == ElementType::Simplex)
+  {
+    AssertThrow(
+      dim == 2,
+      dealii::ExcMessage(
+        "You are trying to create a grid with 3D Simplices. This is currently not supported "
+        "for this application."));
+  }
+
   select_cylinder_type(cylinder_type_string);
 
   switch(cylinder_type)
   {
     case circular:
+    {
+      AssertThrow(
+        element_type == ElementType::Hypercube,
+        dealii::ExcMessage(
+          "You are trying to create a grid with simplex elements for the circular cylinder case, "
+          "which is currently not supported."));
```

Comment:
Why is this not working?

Reply:
I have tried this using `convert_hypercube_to_simplex_mesh()`, since the cylinder case has `hyper_shells`. But I had problems with manifold ids not matching. I didn't invest more time in this and moved on to reading an external grid for the circular case, which works fine.